numba/cuda/tests/cudapy/test_vector_type.py

```diff
@@ -0,0 +1,265 @@
+import os
+
+import numpy as np
+
+from numba.cuda.testing import CUDATestCase
+
+from numba import cuda
+
+if os.environ.get("NUMBA_ENABLE_CUDASIM", 0):
+    vector_types = cuda.vector_types._simulated_vector_types
```

Comment:
Yes. the strange part is the line that follows, the vector types needs to be injected into `cuda` namespace in user domain. I fixed this with importing the vector types and exposing them to `numba.cuda` module in `simulator.__init__.py`.

Reply:
For this import line I think it's fine, because `vector_type` is internal and should never be accessed by user, we are importing the list of vector types here so that we can loop over the vector types and test them programmatically.